src/track/track.cpp

```diff
@@ -1374,18 +1484,48 @@ void Track::updateAudioPropertiesFromStream(
     bool updated = m_record.refMetadata().updateAudioPropertiesFromStream(
             streamInfo);
     m_streamInfo = std::make_optional(std::move(streamInfo));
-    if (!m_pCueInfoImporterPending || m_pCueInfoImporterPending->isEmpty()) {
+
+    bool importBeats = m_pBeatsImporterPending && !m_pBeatsImporterPending->isEmpty();
+    bool importCueInfos = m_pCueInfoImporterPending && !m_pCueInfoImporterPending->isEmpty();
+
+    if (!importBeats && !importCueInfos) {
         // Nothing more to do
         if (updated) {
             markDirtyAndUnlock(&lock);
         }
         return;
     }
-    DEBUG_ASSERT(m_cuePoints.isEmpty());
-    kLogger.debug()
-            << "Finishing deferred import of"
-            << m_pCueInfoImporterPending->size()
-            << "cue(s)";
-    importPendingCueInfosMarkDirtyAndUnlock(
-            &lock);
+
+    auto bpmValue = mixxx::Bpm::kValueUndefined;
+
+    if (importBeats) {
+        kLogger.debug()
+                << "Finishing deferred import of beats";
```

Comment:
Why was it deferred and where?

Reply:
Because the positions depend on the tracks sample rate. Here:
https://github.com/mixxxdj/mixxx/pull/2958/files/2eba4e80bc28361a826d30654858862b591e9bb6#diff-f59a0f669757d180c782c7a8fa0adea0R933-R946

Reply:
OK, so the comment can become: 
"Finishing import because we know now track sample rate and offset"